loader/proto.go

```diff
@@ -751,6 +723,454 @@ func (b *builder) handlePackage() (string, error) {
 	return w.String(), nil
 }
 
+func isGoType(typ string) bool {
+	switch typ {
+	case "bool",
+		"string",
+		"[]byte",
+		"float64",
+		"float32",
+		"int",
+		"int32",
+		"int64",
+		"uint32",
+		"uint64":
+		return true
+	default:
+		return false
+	}
+}
+
+func (b *builder) fromStructToAnnotation(val interface{}) string {
+	w := &strings.Builder{}
+
+	s := reflect.TypeOf(val)
+	v := reflect.ValueOf(val)
+
+	for i := 0; i < s.NumField(); i++ {
+		name := s.Field(i).Name
+		value := v.FieldByName(name)
+		if !isNilOrEmpty(value.Interface()) {
+			k := value.Kind()
+			switch {
+			case k >= reflect.Int && k <= reflect.Int64,
+				k == reflect.Float32 || k == reflect.Float64,
+				k >= reflect.Uint && k <= reflect.Uint64,
+				k == reflect.String,
+				k == reflect.Bool:
```

Comment:
I'm not sure I get this logic. Is this to check if the kind is a builtin type?

Reply:
Another option would be to handle map/ptr/slice separately, reject/ignore specific types like array, and then make this the "default" case.